### PR TITLE
fix: various typos

### PR DESCRIPTION
**Pull request by NathanBaulch**
Just thought I'd contribute some typo fixes that I stumbled on. Nothing controversial (hopefully).

Use the following command to get a quick summary of the specific corrections made:
```shell
git diff HEAD^! --word-diff-regex='\w+' -U0 \
  | grep -E '\[\-.*\-\]\{\+.*\+\}' \
  | sed -r 's/.*\[\-(.*)\-\]\{\+(.*)\+\}.*/\1 \2/' \
  | sort | uniq -c | sort -n
```

FWIW, the top typos are:
* contructor (3)
* occured (2)
* reuest (2)
* shoudl (1)
* afterwords (1)
* availabile (1)
* concatanete (1)
* existant (1)
* isable (1)
* statu (1)
* undinci (1)

### CODE REVIEW

File: docs/docs/api/MockPool.md [hunk 1]
```diff
@@ -81,7 +81,7 @@ By default, `reply` and `replyWithError` define the behaviour for the first matc
 
 ### Return: `MockScope`
 
-A `MockScope` is associated with a single `MockInterceptor`. With this, we can configure the default behaviour of a intercepted reply.
+A `MockScope` is associated with a single `MockInterceptor`. With this, we can configure the default behaviour of an intercepted reply.
```
Comment (szepeviktor) [inline docs/docs/api/MockPool.md:84]: @NathanBaulch I also do random PR-s with `typos`.
How do you spot wrong indefinite articles?

Comment (NathanBaulch) [reply docs/docs/api/MockPool.md:84]: IntelliJ highlights them. I usually only fix them if the PR I'm working on isn't getting too out of hand.

Comment (szepeviktor) [reply docs/docs/api/MockPool.md:84]: https://www.jetbrains.com/help/idea/grammar.html

**Review by KhafraDev — CHANGES_REQUESTED** [review]
Most of the changes to lib/web should be reverted.

**Comment by NathanBaulch** [comment]
Is `lib/web` an upstream package @KhafraDev?

**Comment by KhafraDev** [comment]
The comments come from the various whatwg specs we implement.

File: lib/llhttp/constants.js
Comment (metcoder95) [inline lib/llhttp/constants.js:1]: Let's rever this one as well. I know the typo is wrong but are statuses we depend upon on runtime

Comment (NathanBaulch) [reply lib/llhttp/constants.js:1]: Done.

**Review by metcoder95 — APPROVED** [review]

**Review by Uzlopak — APPROVED** [review]
LGTM

**Pull request merged**